File: src/sorting/bubbleSort.c
#include <stdio.h>
#include <stdlib.h>
#include "./_common_/common.h"

void bubble_sort(int* arr, const int size)
{
    for (int i = 0; i < size - 1; i++)
        for (int j = 0; j < size - 1 - i; j++)
            if (arr[j] > arr[j + 1])
                swap(&arr[j], &arr[j + 1]);
}

int main(void)
{
    int n;

    scanf("%d", &n);
    int* arr = (int*)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    print_array(arr, n);
    bubble_sort(arr, n);
    print_array(arr, n);

    free(arr);
    return 0;
}

/*
Input:
5
5 4 3 2 1

Output:
5 4 3 2 1
1 2 3 4 5
*/

File: src/sorting/bubble-sort.js
function bubbleSort(arr) {
    let size = arr.length;
    for (let i = 0; i < size - 1; i++) {
        for (let j = 0; j < size - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) {
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
            }
        }
    }

    return arr;
}

const arr = [955, 393, 719, 189, 682, 685, 825, 939, 778, 979, 10, 523];
let result = "Before: ";
for (let i in arr) {
    result = result.concat(`${arr[i]} `);
}
const rt = bubbleSort(arr);
result = result.concat('\nAfter: ');
for (let i in rt) {
    result = result.concat(`${rt[i]} `);
}
console.log(result);


/*
Output:
Before: 955 393 719 189 682 685 825 939 778 979 10 523 
After: 10 189 393 523 682 685 719 778 825 939 955 979
*/